File: backend/src/main/java/gruppe4/tessera/model/Transaction.java
package gruppe4.tessera.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Table(name = "transactions")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "transaction_id") // ID will be used as primarykey and will be autoincremented
    private Integer id;

    private Integer postId, buyerId, sellerId;

    @JsonProperty("transaction_id")
    public Integer getId(){
        return id;
    }

}

File: frontend/tessera-webapp/src/DataTypes/Transaction.ts
export { Transaction };

class Transaction {
  sellerID: number;
  buyerID: number;
  postID: number;

  constructor(sellerID: number, buyerID: number, postID: number) {
    this.sellerID = sellerID;
    this.buyerID = buyerID;
    this.postID = postID;
  }

  getTransactionData() {
    return {
      sellerID: this.sellerID,
      buyerID: this.buyerID,
      postID: this.postID
    }
  }
}
